src/components/Table/GridTable.stories.tsx

```diff
@@ -219,24 +190,26 @@ export function NestedCardsThreeLevels() {
   return <NestedCards rows={rowsWithHeader} sorting={{ on: "client", initial: [0, "ASC"] }} />;
 }
 
+function deepCount(rows: GridDataRow<any>[]): number {
+  return rows.map((row) => 1 + deepCount(row.children ?? [])).reduce((a, b) => a + b, 0);
+}
+
 export function NestedCardsThreeLevelsVirtualizedAtScale() {
+  const rows = useMemo(() => [header, ...makeNestedRows(500)], []);
   return (
     <div css={Css.df.fdc.vh100.$}>
-      Rendering {rows.length * 500} rows virtualized
-      <NestedCards rows={[header, ...zeroTo(100).flatMap(() => rows)]} as={"virtual"} />
+      Rendering {deepCount(rows)} rows virtualized
+      <NestedCards rows={rows} as="virtual" />
     </div>
   );
 }
 
 export function NestedCardsThreeLevelsVirtualizedAtScaleSorted() {
+  const rows = useMemo(() => [header, ...makeNestedRows(500)], []);
```

Comment:
@chr1sjf0x fyi I refactored these "at scale" stories while lazy-izing their content, and fwiw they're pretty snappy, like now they have ~5000 some rows and sorting is still ~instant. Pretty neat.

Reply:
Awesome 👌